handler/src/main/java/io/netty/handler/ssl/OpenSslContext.java

```diff
@@ -501,8 +524,52 @@ public OpenSslEngine remove(long ssl) {
         public void add(OpenSslEngine engine) {
             engines.put(engine.sslPointer(), engine);
         }
+
+        @Override
+        public OpenSslEngine get(long ssl) {
+            return engines.get(ssl);
+        }
     }
 
+    static void setKeyMaterial(long ctx, X509Certificate[] keyCertChain, PrivateKey key, String keyPassword)
+            throws SSLException {
+         /* Load the certificate file and private key. */
+        long keyBio = 0;
+        long keyCertChainBio = 0;
+
+        try {
+            keyCertChainBio = toBIO(keyCertChain);
+            keyBio = toBIO(key);
+
+            if (!SSLContext.setCertificateBio(
+                    ctx, keyCertChainBio, keyBio, keyPassword, SSL.SSL_AIDX_RSA)) {
+                long error = SSL.getLastErrorNumber();
```

Comment:
`setCertificateBio` calls `ERR_clear_error` and throws an exception ... should this method just be void and throw exceptions ... or rely on the java code to get/clear the error queue?